sklearn/utils/_sorting.pxd

```diff
@@ -0,0 +1,19 @@
+from ._typedefs cimport DTYPE_t, ITYPE_t
+
+from cython cimport floating
+
+cdef int partition_node_indices(
+    DTYPE_t *data,
+    ITYPE_t *node_indices,
+    ITYPE_t split_dim,
+    ITYPE_t split_index,
+    ITYPE_t n_features,
+    ITYPE_t n_points
+) nogil except -1
+
+
+cdef int simultaneous_sort(
```

Comment:
I think we can have this one in `utils._sorting`, as it is used between `metrics` and `neighbors`. This is also a drop in replacement for the simultaneous sort in trees as seen in: https://github.com/scikit-learn/scikit-learn/pull/22868